main.nf

```diff
@@ -20,103 +20,357 @@
+    }
+
+    //
+    // WORKFLOW: Run pipeline
+    //
+    SAREK(samplesheet,
+        allele_files,
+        bcftools_annotations,
+        bcftools_annotations_tbi,
+        bcftools_header_lines,
+        cf_chrom_len,
+        chr_files,
+        cnvkit_reference,
+        dbsnp,
+        dbsnp_tbi,
+        dbsnp_vqsr,
+        dict,
+        fasta,
+        fasta_fai,
+        gc_file,
+        germline_resource,
+        germline_resource_tbi,
+        index_alignement,
+        intervals_and_num_intervals,
+        intervals_bed_combined,
+        intervals_bed_combined_for_variant_calling,
+        intervals_bed_gz_tbi_and_num_intervals,
+        intervals_bed_gz_tbi_combined,
+        intervals_for_preprocessing,
+        known_indels_vqsr,
+        known_sites_indels,
+        known_sites_indels_tbi,
+        known_sites_snps,
+        known_sites_snps_tbi,
+        known_snps_vqsr,
+        loci_files,
+        mappability,
+        msisensorpro_scan,
+        ngscheckmate_bed,
+        pon,
+        pon_tbi,
+        rt_file,
+        sentieon_dnascope_model,
+        snpeff_cache,
+        vep_cache,
+        vep_cache_version,
+        vep_extra_files,
+        vep_fasta,
+        vep_genome,
+        vep_species
+    )
+
+    emit:
+    multiqc_report = SAREK.out.multiqc_report // channel: /path/to/multiqc_report.html
+}
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    NAMED WORKFLOW FOR PIPELINE
+    RUN MAIN WORKFLOW
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { SAREK } from './workflows/sarek'
+workflow {
 
-// WORKFLOW: Run main nf-core/sarek analysis pipeline
-workflow NFCORE_SAREK {
-    SAREK ()
+    main:
+
+    //
+    // SUBWORKFLOW: Run initialisation tasks
+    //
+    PIPELINE_INITIALISATION(
+        params.version,
+        params.help,
+        params.validate_params,
+        params.monochrome_logs,
+        args,
+        params.outdir,
+        params.input
+    )
+
+    //
+    // WORKFLOW: Run main workflow
+    //
+    NFCORE_SAREK(PIPELINE_INITIALISATION.out.samplesheet)
+
+    //
+    // SUBWORKFLOW: Run completion tasks
+    //
+    PIPELINE_COMPLETION(
+        params.email,
+        params.email_on_fail,
+        params.plaintext_email,
+        params.outdir,
+        params.monochrome_logs,
+        params.hook_url,
+        NFCORE_SAREK.out.multiqc_report
+    )
 }
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    RUN ALL WORKFLOWS
+    FUNCTIONS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-// WORKFLOW: Execute a single named workflow for the pipeline
-// See: https://github.com/nf-core/rnaseq/issues/619
-workflow {
-    NFCORE_SAREK ()
+//
+// Get attribute from genome config file e.g. fasta
+//
+
+def getGenomeAttribute(attribute) {
```

Comment:
It doesn't matter if `getGenomeAttribute `is in the `main.nf` or in `subworkflows/nf-core/utils_nextflow_pipeline/main.nf` ?

Reply:
@ramprasadn and I noticed some issues with the `getGenomeAttribute` from `subworkflows/nf-core/utils_nextflow_pipeline/main.nf` , so we're trying out a local one in the `main.nf`